### ISSUE TITLE
authentication: Websocket connection unauthorized?

### ISSUE THREAD

**Issue by bernhardreiter** [dddomodossola/remi#152]
When requesting authentication like
```python
   start(MyApp, debug=True, start_browser=False,
          port=9091, websocket_port=50000,
          username='me', password='letmein')
```

There is a basci auth header send with the http request to the http port,
but I do not seen any authentication on the websocket port. What prevents
a malicious client from initiating a websocket connection?

**Comment by nzjrs** [comment]
> What prevents a malicious client from initiating a websocket connection?

nothing.

**Comment by nzjrs** [comment]
I have always tried to emphasize remi as a desktop ui alternative, and not a toolkit for people wanting to make websites without learning javascript. In some sense I regret adding authentication at all, as it helps to give such an incorrect impression.

**Comment by bernhardreiter** [comment]
@nzjrs the other solution is to be very explicit about this and add a huge warning.

Remi as a webapp building application would be attractive so I guess many people will get the idea anyway. In addition I believe that security aspects are interesting for desktop ui use as well.

**Comment by bernhardreiter** [comment]
Maybe adding basic-auth to the websocket connection would be a reasonable next step.

**Comment by bernhardreiter** [comment]
Here is a helpful link I've found https://devcenter.heroku.com/articles/websocket-security

**Comment by simplynail** [comment]
What about using remi as gui/client all the way (with callbacks just modifying the US or sending requests to server) and on the server side having only a Rest API that responds? Wouldn't it solve it?

**Comment by simplynail** [comment]
I meant 'modifying UI'

**Comment by nzjrs** [comment]
That would be a rewrite. Might as well wait for server sent events to be more popular.

**Comment by simplynail** [comment]
No, I mean to have remi gui app running as local server on desktop/client and somewhere on the Web having eg Flask receiving queries and sending json responses

**Comment by nzjrs** [comment]
What problem are you trying to solve? I'm sorry I don't see the connection
to this bug.

**Comment by simplynail** [comment]
As You stated before, remi doesn't aim to be a web framework and thus this is not actually a bug. So You are right that this doesn't solve the bug, but maybe it can be sufficient workaround in some cases. That's it.
